### PR TITLE
Latest bindings for VSCode.LanguageServer.fs

### LINKED ISSUE #49: Update bindings for vscode-languageclient 8.0
**Describe the bug**

We need to update our bindings. There were some [breaking changes in 8.0](https://github.com/microsoft/vscode-languageserver-node#3170-protocol-800-json-rpc-800-client-and-800-server)  .@chrissimon-au did some of that work over in https://github.com/dev-cycles/contextive/commit/a3f7ed84f5c22d5ffa899dc033ee6076d8bc54ff

### PR DESCRIPTION
**Pull request by TheAngryByrd**
Closes #49

### CODE REVIEW

File: .config/dotnet-tools.json [hunk 1]
```diff
@@ -1,30 +1,30 @@
...
+      ]
+    },
+    "fake-cli": {
+      "version": "5.20.3",
+      "commands": [
+        "fake"
+      ]
+    },
+    "fable": {
+      "version": "3.7.22",
```
Comment (TheAngryByrd) [inline .config/dotnet-tools.json:18]: Needed a newer version of Fable.

File: build.fsx [hunk 2]
```diff
@@ -7,21 +7,30 @@ open Fake.JavaScript
...
 open Fake.IO
 
-let runFable args = 
+module BuildParameters =
+    let common (defaults:MSBuild.CliArguments) =
+        { defaults with
+            Verbosity = Some MSBuildVerbosity.Minimal
+            // see https://github.com/fsprojects/FAKE/issues/2515#issue-622856864
+            // this avoids Fake to crasing while trying to load MSBuild 15.0:
+            DisableInternalBinLog = true
```
Comment (TheAngryByrd) [inline build.fsx:16]: Sidelines fake msbuild binlog issue

File: global.json [hunk 3]
```diff
@@ -0,0 +1,6 @@
+{
+  "sdk": {
+    "version": "6.0.0",
```
Comment (TheAngryByrd) [inline global.json:3]: Fable 3 only works on 6.x.  We could goto 4 but I don't want to go for touching more than I need to.

File: src/Fable.Import.VSCode.LanguageServer.fs [hunk 4]
```diff
@@ -278,8 +279,7 @@ module rec LanguageServer =
         member __.error(message: string, ?data: obj): unit = jsNative
         member __.needsStart(): bool = jsNative
         member __.needsStop(): bool = jsNative
-        member __.onReady(): Promise<unit> = jsNative
-        member __.start(): Disposable = jsNative
+        member __.start() : Promise<unit> = jsNative
```
Comment (TheAngryByrd) [inline src/Fable.Import.VSCode.LanguageServer.fs:282]: The breaking change mentioned in #49

File: src/Fable.Import.VSCode.LanguageServer.fs [hunk 5]
```diff
@@ -256,7 +257,7 @@ module rec LanguageServer =
         abstract detached: bool with get, set
 
     type ServerOptions =
-        U7<Executable, obj, obj, obj, obj, NodeModule, (unit -> Promise<U4<ChildProcess, StreamInfo, Client.MessageTransports, ChildProcessInfo>>)>
+        U5<Executable, {| run: Executable; debug: Executable |}, {| run: NodeModule; debug: NodeModule |}, NodeModule, (unit -> Promise<U4<ChildProcess, StreamInfo, Client.MessageTransports, ChildProcessInfo>>)>
```
Comment (TheAngryByrd) [inline src/Fable.Import.VSCode.LanguageServer.fs:260]: Not strictly necessary for this but I think this will be useful for other things later.

**Review by baronfel — APPROVED** [review]
LGTM! I opened this diff expecting to go through a lot of API changes, but then saw that most of that 2k were in yarn.lock 😅

**Comment by TheAngryByrd** [comment]
> LGTM! I opened this diff expecting to go through a lot of API changes, but then saw that most of that 2k were in yarn.lock 😅

Yeah nothing I can really do about that :(

**Pull request merged**